### PR TITLE
[KOGITO-1248] Add casemgmt tests

### PR DESCRIPTION
**Pull request by ruromero**
Signed-off-by: ruromero <rromerom@redhat.com>

Fix https://issues.redhat.com/browse/KOGITO-1248

Added case management processes from JBPM7.x and implemented visitors and missing methods for DynamicNodes, MilestoneNodes and StateNodes

### CODE REVIEW

File: jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/AbstractNodeVisitor.java [hunk 1]
```diff
@@ -48,15 +48,16 @@
...
-    protected static final String METADATA_TRIGGER_TYPE = "TriggerType";
-    protected static final String METADATA_TRIGGER_MAPPING = "TriggerMapping";
-    protected static final String METADATA_MAPPING_VARIABLE = "MappingVariable";
-    protected static final String METADATA_EVENT_TYPE = "EventType";
+    public static final String METADATA_TRIGGER_REF = "TriggerRef";
+    public static final String METADATA_MESSAGE_TYPE = "MessageType";
+    public static final String METADATA_TRIGGER_TYPE = "TriggerType";
+    public static final String METADATA_TRIGGER_MAPPING = "TriggerMapping";
+    public static final String METADATA_MAPPING_VARIABLE = "MappingVariable";
+    public static final String METADATA_EVENT_TYPE = "EventType";
```
Comment (evacchi) [inline jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/AbstractNodeVisitor.java:60]: if these are public maybe we should move them to some more descriptive class name. Like Messages or something like that?

File: api/kogito-api/src/main/java/org/kie/api/definition/process/Node.java [hunk 2]
```diff
@@ -72,9 +72,9 @@
     /**
      * The NodeContainer this Node lives in.
      *
-     * @return the NodeContainer
+     * @return the parent NodeContainer
      */
-    NodeContainer getNodeContainer();
+    NodeContainer getParentContainer();
```
Comment (evacchi) [inline api/kogito-api/src/main/java/org/kie/api/definition/process/Node.java:77]: could it be mis-read? I was thinking this was pointing to the "parent" of this node (i.e. another node).

Comment (ruromero) [reply api/kogito-api/src/main/java/org/kie/api/definition/process/Node.java:77]: I renamed it because some nodes have their own nodeContainer. Besides, whenever this method is used is assigned to a variable called `parent` and the attribute returned is `parentNodeContainer`.

File: jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/AbstractNodeVisitor.java (hunk 1 above)
Comment (ruromero) [reply jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/AbstractNodeVisitor.java:60]: Do you think a Metadata interface would be better? I thought putting these constants in the "owner" visitors or in the common ancestor if they are shared.

**Comment by ruromero** [comment]
jenkins retest this

File: jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/core/impl/NodeImpl.java [hunk 3]
```diff
@@ -245,7 +245,7 @@ public Connection getTo() {
         return getOutgoingConnections(org.jbpm.workflow.core.Node.CONNECTION_DEFAULT_TYPE);
     }
 
-    public NodeContainer getNodeContainer() {
+    public NodeContainer getParentContainer() {
         return parentNodeContainer;
```
Comment (MarianMacik) [inline jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/core/impl/NodeImpl.java:249]: Hmm, what about calling it `getParentNodeContainer()`? This way it will be consistent, i.e. `getNodeContainer()` for getting this node's container and `getParentNodeContainer()` for getting the node container of this node parent :) Wdyt?

File: kogito-codegen/src/test/java/org/kie/kogito/codegen/process/ProcessGenerationTest.java [hunk 4]
```diff
@@ -194,7 +197,7 @@ private static void assertActions(Node eNode, ExtendedNodeImpl expected, Extende
             try {
                 if (expected.getActions(actionType) == null) {
                     assertNull(current.getActions(actionType));
-                } else {
+                } else if (!EVENT_NODE_ENTER.equals(actionType) && !EVENT_NODE_EXIT.equals(actionType)){
```
Comment (MarianMacik) [inline kogito-codegen/src/test/java/org/kie/kogito/codegen/process/ProcessGenerationTest.java:200]: So in this case the actionType key is not present in a map? In this case we should probably assert for null in the previous if condition, right?

File: kogito-codegen/src/test/resources/org/kie/kogito/codegen/process/process-generation-test.skip.txt [hunk 5]
```diff
@@ -1,3 +1,6 @@
+cases/ScriptRoleAssignmentCase.bpmn2
+cases/UserTaskWithStageCase.bpmn2
+cases/UserTaskWithStageCaseAutoStart.bpmn2
```
Comment (MarianMacik) [inline kogito-codegen/src/test/resources/org/kie/kogito/codegen/process/process-generation-test.skip.txt:3]: What is preventing us from testing these 3 cases specifically? You added many more and these are (presumably) tested.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/core/impl/NodeImpl.java (hunk 3 above)
Comment (ruromero) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/core/impl/NodeImpl.java:249]: What about just `getContainer` and `getParentContainer` ? FYI not all nodes have their own container, only `CompositeNode` nodes have it.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/BoundaryEventNodeFactory.java [hunk 6]
```diff
@@ -118,24 +115,4 @@ public BoundaryEventNodeFactory eventType(String eventTypePrefix, String eventTy
...
-    }
-
-    public BoundaryEventNodeFactory timeDuration(String timeDuration) {
-        eventType(EVENT_TYPE_TIMER, timeDuration);
-        return metaData(METADATA_TIME_DURATION, timeDuration);
-    }
-
-    public BoundaryEventNodeFactory cancelActivity(boolean cancelActivity) {
-        return metaData(METADATA_CANCEL_ACTIVITY, cancelActivity);
-    }
```
[the remark this answers is not in the document]
Comment (ruromero) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/BoundaryEventNodeFactory.java:140]: The main reason is that Metadata attributes are copied in the VisitMetadata method so this seems redundant.

File: kogito-codegen/src/test/resources/org/kie/kogito/codegen/process/process-generation-test.skip.txt (hunk 5 above)
Comment (ruromero) [reply kogito-codegen/src/test/resources/org/kie/kogito/codegen/process/process-generation-test.skip.txt:3]: The two UserTaskWithStageCase processes depend on custom model definition and are ignored like in other processes.
The ScriptRoleAssignmentCase should be removed once the case model supports caseAssignments or something equivalent and the process is updated.

File: kogito-codegen/src/test/java/org/kie/kogito/codegen/process/ProcessGenerationTest.java (hunk 4 above)
Comment (MarianMacik) [reply kogito-codegen/src/test/java/org/kie/kogito/codegen/process/ProcessGenerationTest.java:200]: As discussed, this is ok.

**Comment by ruromero** [comment]
I have done all the changes

File: jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/MilestoneNodeVisitor.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    private static final String NODE_KEY = "milestoneNode";
+
+    @Override
+    protected String getNodeKey() {
+        return NODE_KEY;
+    }
+
+    @Override
+    public void visitNode(String factoryField, Node node, BlockStmt body, VariableScope variableScope, ProcessMetaData metadata) {
+        MilestoneNode milestoneNode = (MilestoneNode) node;
```
Comment (tiagodolphine) [inline jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/MilestoneNodeVisitor.java:42]: same here, to avoid the cast it could be used generics, maybe on AbstractNodeVisitor<T>

File: jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/core/Node.java [hunk 8]
```diff
@@ -28,7 +28,7 @@
  */
 public interface Node extends org.kie.api.definition.process.Node, Contextable, Serializable {
 
-    static final String CONNECTION_DEFAULT_TYPE = "DROOLS_DEFAULT";
+    String CONNECTION_DEFAULT_TYPE = "DROOLS_DEFAULT";
```
Comment (tiagodolphine) [inline jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/core/Node.java:31]: why the `static final` was removed? Anyway, if it is not a constant anymore you can change the name to camel case.

File: api/kogito-api/src/main/java/org/kie/api/definition/process/Node.java (hunk 2 above)
Comment (tiagodolphine) [reply api/kogito-api/src/main/java/org/kie/api/definition/process/Node.java:77]: Only `CompositeNode` has its own nodeContainer? I mean it seems it is returning itself. 
Why not just overriding the original `getNodeContainer` on  `CompositeNode`, or whatever node that has a different behavior other than returning the parent.

File: kogito-codegen/src/test/resources/org/kie/kogito/codegen/process/process-generation-test.skip.txt (hunk 5 above)
Comment (MarianMacik) [reply kogito-codegen/src/test/resources/org/kie/kogito/codegen/process/process-generation-test.skip.txt:3]: Ah, I see. And about the custom model - is there a plan to support it as well? Just curious.

File: jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/CompositeContextNodeVisitor.java [hunk 9]
```diff
@@ -71,6 +73,8 @@ public void visitNode(String factoryField, Node node, BlockStmt body, VariableSc
             visitVariableScope(getNodeId(node), variableScopeNode, body, new HashSet<>());
         }
 
+        visitCustomFields(compositeContextNode).forEach(body::addStatement);
```
[the remark this answers is not in the document]
Comment (MarianMacik) [reply jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/CompositeContextNodeVisitor.java:76]: Probably the best option would be to pass body as another argument as with other visitors.

File: api/kogito-api/src/main/java/org/kie/api/definition/process/Node.java (hunk 2 above)
Comment (MarianMacik) [reply api/kogito-api/src/main/java/org/kie/api/definition/process/Node.java:77]: Ah, I see now. `CompositeNode` has another `NodeContainer` named `nodeContainer` but this, until now, wasn't returned by any of the `CompositeNode` methods. So by calling `getNodeContainer()` the `NodeContainer` from the `NodeImpl` parent was returned and there was no option to obtain the `NodeContainer` of `CompositeNode` using a getter. Now, the `getParentContainer()` will return the one-level-above container (the one from `NodeImpl`) and `getNodeContainer()` will return the one from `CompositeNode`. In practice, these nodes can be nested into each other many times, but now it is possible to obtain just the container with nodes inside ("belonging to") the `CompositeNode` and the container which `CompositeNode` is in separately.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/BoundaryEventNodeFactory.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (MarianMacik) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/BoundaryEventNodeFactory.java:140]: I see, so eventTypes and metaData are "copied" when bpmn is traversed anyway.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/core/impl/NodeImpl.java (hunk 3 above)
Comment (MarianMacik) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/core/impl/NodeImpl.java]: Yes, +1.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/core/Node.java (hunk 8 above)
Comment (ruromero) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/core/Node.java:31]: On interfaces, every field is defined as static final and public so this is redundant.

**Comment by ruromero** [comment]
> Only CompositeNode has its own nodeContainer? I mean it seems it is returning itself.
Why not just overriding the original getNodeContainer on CompositeNode, or whatever node that has a different behavior other than returning the parent.

Because it also has to return the parentNode for some generic methods. It's not only the `CompositeNode` but also it's descendants.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/core/Node.java (hunk 8 above)
Comment (tiagodolphine) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/core/Node.java:31]: ooh this is an interface, yep you are right!

File: jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/CompositeContextNodeVisitor.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (ruromero) [reply jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/CompositeContextNodeVisitor.java:76]: > why not passing the function,`body::addStatement` as a parameter to the `visitCustomFields` method and apply it on the method implementation, I mean to keep the same approach of other visit methods, or just change the method name to `getCustomFields` ?

That will make it look very different from the other visit methods.

> Probably the best option would be to pass body as another argument as with other visitors.

I wanted to avoid passing an object to be modified as an argument. This is part of some refactoring I did everywhere that didn't require too big changes, like in visitNodes.

File: jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/DynamicNodeVisitor.java [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Override
+    protected String getDefaultName() {
+        return "Dynamic";
+    }
+
+    @Override
+    public Stream<MethodCallExpr> visitCustomFields(CompositeContextNode node) {
+        Collection<MethodCallExpr> methods = new ArrayList<>();
+        DynamicNode dynamicNode = (DynamicNode) node;
```
[the remark this answers is not in the document]
Comment (ruromero) [reply jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/DynamicNodeVisitor.java]: I have refactored all the visitor classes to make them use generics, I think they look much better. Thanks for the suggestion.

File: jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/AbstractCompositeNodeVisitor.java [hunk 11]
```diff
@@ -17,21 +17,27 @@
...
-        for (Node node : nodes) {
-            AbstractNodeVisitor visitor = nodesVisitors.get(node.getClass());
+    @Override
+    public void visitNode(T node, BlockStmt body, VariableScope variableScope, ProcessMetaData metadata) {
+        super.visitNode(node, body, variableScope, metadata);
+    }
+
+    protected <U extends Node> void visitNodes(String factoryField, U[] nodes, BlockStmt body, VariableScope variableScope, ProcessMetaData metadata) {
+        for (U node : nodes) {
+            AbstractNodeVisitor<U> visitor = (AbstractNodeVisitor<U>) nodesVisitors.get(node.getClass());
```
Comment (MarianMacik) [inline jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/AbstractCompositeNodeVisitor.java:40]: Generics now works but just because here we use unsafe casting. Without it the types in the call on line 44 `visitor.visitNode(factoryField, node, body, variableScope, metadata);` wouldn't match as `U[] nodes` is always array of Node (passed from the visitNode method) so U is Node but then we are passing this `node` variable of type `Node` to that method call on line 44. But given we did unsafe casting here, compiler is happy and it is ok after type erasure as during runtime the signature of abstractNodeVisitor visitNode() method has `Node`.

In short, I am not sure if this helped the hierarchy :) We also use raw types for Visitors in ProcessVisitor.java file.

File: jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/ProcessVisitor.java [hunk 12]
```diff
@@ -103,6 +105,9 @@ public ProcessVisitor(ClassLoader contextClassLoader) {
         this.nodesVisitors.put(CompositeContextNode.class, new CompositeContextNodeVisitor(nodesVisitors));
```
Comment (MarianMacik) [inline jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/ProcessVisitor.java:105]: Here we are using raw types for example.

Same issue with unsafe calls now appeared in `visitNodes` method as well.

**Comment by tiagodolphine** [comment]
@cristianonicolai would you like to take a look at the PR, or are we ok to merge it?
Thanks.

File: jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/EventSubprocessNodeVisitor.java [hunk 13]
```diff
@@ -15,16 +15,27 @@
...
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Map;
+import java.util.stream.Stream;
 
-public class EventSubprocessNodeVisitor extends CompositeContextNodeVisitor {
+import static org.jbpm.ruleflow.core.factory.EventSubProcessNodeFactory.METHOD_EVENT;
+import static org.jbpm.ruleflow.core.factory.EventSubProcessNodeFactory.METHOD_KEEP_ACTIVE;
+
+public class EventSubprocessNodeVisitor extends CompositeContextNodeVisitor<EventSubProcessNode> {
```
Comment (cristianonicolai) [inline jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/EventSubprocessNodeVisitor.java:34]:
```suggestion
public class EventSubProcessNodeVisitor extends CompositeContextNodeVisitor<EventSubProcessNode> {
```

File: jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/MilestoneNodeVisitor.java [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+    private static final String NODE_KEY = "milestoneNode";
+
+    @Override
+    protected String getNodeKey() {
+        return NODE_KEY;
+    }
+
+    @Override
+    public void visitNode(String factoryField, MilestoneNode node, BlockStmt body, VariableScope variableScope, ProcessMetaData metadata) {
+        body.addStatement(getAssignedFactoryMethod(factoryField, MilestoneNodeFactory.class, getNodeId(node), NODE_KEY,new LongLiteralExpr(node.getId())))
```
Comment (cristianonicolai) [inline jbpm/jbpm-flow-builder/src/main/java/org/jbpm/compiler/canonical/MilestoneNodeVisitor.java:41]:
```suggestion
        body.addStatement(getAssignedFactoryMethod(factoryField, MilestoneNodeFactory.class, getNodeId(node), NODE_KEY, new LongLiteralExpr(node.getId())))
```

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/process/ProcessCodegen.java [hunk 15]
```diff
@@ -84,7 +84,7 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ProcessCodegen.class);
 
     private static final SemanticModules BPMN_SEMANTIC_MODULES = new SemanticModules();
-    public static final Set<String> SUPPORTED_BPMN_EXTENSIONS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(".bpmn", ".bpmn2")));
+    public static final Set<String> SUPPORTED_BPMN_EXTENSIONS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(".bpmn", ".bpmn2", ".bpmn-cm")));
```
[the remark this answers is not in the document]
Comment (ruromero) [reply kogito-codegen/src/main/java/org/kie/kogito/codegen/process/ProcessCodegen.java:87]: bpmn-cm is a format introduced in jbpm7 for case modeler. It is compatible with bpmn2 and as we are adding functionalities for case management processes I think it's nice to have. Otherwise, processes created with the case modeler jbpm7.x will have to be renamed to work on Kogito.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/DynamicNodeFactory.java [hunk 16]
```diff
@@ -25,6 +25,10 @@
 
 public class DynamicNodeFactory extends CompositeContextNodeFactory {
 
+	public static final String METHOD_LANGUAGE = "language";
```
[the remark this answers is not in the document]
Comment (ruromero) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/DynamicNodeFactory.java:28]: It's just a matter of taste, I considered the owner of the method should have the constant that will be used by the visitor to make it easier to maintain and remove/update as the methods are defined here.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/EventSubProcessNodeFactory.java [hunk 17]
```diff
@@ -16,13 +16,16 @@
...
 
+import org.jbpm.process.core.event.EventTypeFilter;
 import org.jbpm.ruleflow.core.RuleFlowNodeContainerFactory;
 import org.jbpm.workflow.core.NodeContainer;
 import org.jbpm.workflow.core.node.CompositeContextNode;
 import org.jbpm.workflow.core.node.EventSubProcessNode;
 
 public class EventSubProcessNodeFactory extends CompositeContextNodeFactory {
 
+    public static final String METHOD_KEEP_ACTIVE = "keepActive";
```
[the remark this answers is not in the document]
Comment (ruromero) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/EventSubProcessNodeFactory.java:27]: same

File: jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/EventSubProcessNodeFactory.java [hunk 18]
```diff
@@ -32,4 +35,16 @@ public EventSubProcessNodeFactory(RuleFlowNodeContainerFactory nodeContainerFact
     protected CompositeContextNode createNode() {
         return new EventSubProcessNode();
     }
+
+    public EventSubProcessNodeFactory keepActive(boolean keepActive) {
```
[the remark this answers is not in the document]
Comment (ruromero) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/EventSubProcessNodeFactory.java:39]: It is used in the generated code. See `EventSubprocessNodeVisitor#visitCustomFields`

File: jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/EventSubProcessNodeFactory.java [hunk 19]
```diff
@@ -32,4 +35,16 @@ public EventSubProcessNodeFactory(RuleFlowNodeContainerFactory nodeContainerFact
...
+        ((EventSubProcessNode) getCompositeNode()).setKeepActive(keepActive);
+        return this;
+    }
+
+    public EventSubProcessNodeFactory event(String event) {
```
[the remark this answers is not in the document]
Comment (ruromero) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/EventSubProcessNodeFactory.java:44]: same

File: jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/MilestoneNodeFactory.java [hunk 20]
```diff
@@ -23,6 +23,9 @@
 
 public class MilestoneNodeFactory extends StateBasedNodeFactory {
 
+    public static final String METHOD_CONSTRAINT = "constraint";
```
[the remark this answers is not in the document]
Comment (ruromero) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/MilestoneNodeFactory.java:26]: same as for the `EventSubProcessNodeFactory` I consider factories are the owners of the methods and these literals should be here.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/StateBasedNodeFactory.java [hunk 21]
```diff
@@ -24,6 +24,8 @@
 
 public abstract class StateBasedNodeFactory extends ExtendedNodeFactory {
 
+    public static final String METHOD_TIMER = "timer";
```
[the remark this answers is not in the document]
Comment (ruromero) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/StateBasedNodeFactory.java:27]: They're not used because timers have not yet been added to `StateBasedNodeVisitor` and its hierarchy. However as this method is not new I decided to add it. I can remove it or try to add timers to `WorkItemNode`, `RuleSetNode`, `SubProcessNode`, `MilestoneNode` and `HumanTaskNode`

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/process/ProcessCodegen.java (hunk 15 above)
[the remark this answers is not in the document]
Comment (cristianonicolai) [reply kogito-codegen/src/main/java/org/kie/kogito/codegen/process/ProcessCodegen.java:87]: @ruromero you do have a point but I believe that it should be discussed with the team to understand our overall strategy around it. At least to me, it is not clear if we want to promote that with Kogito.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/StateBasedNodeFactory.java (hunk 21 above)
[the remark this answers is not in the document]
Comment (cristianonicolai) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/factory/StateBasedNodeFactory.java:27]: np to leave it then

**Comment by cristianonicolai** [comment]
@ruromero not sure if the failures on kogito-examples are related, you might need to rebase your PR, please take a look.
